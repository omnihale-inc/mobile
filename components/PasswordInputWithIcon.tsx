import { ReactNode, useState } from "react";
import { StyleSheet, TextInput } from "react-native";

import { View } from "./Themed";
import EyeInvisibleIcon from "./icons/EyeInvisible";
import EyeVisibleIcon from "./icons/EyeVisible";

const PasswordInputWithIcon = ({
  value,
  onChangeText,
  icon,
  placeholder
}: {
  value?: string;
  onChangeText?: (value: string) => void;
  icon: (props?: object) => ReactNode;
  secureText?: boolean;
  placeholder?: string;
}) => {
  const [passwdIsDisplayed, setPasswdIsDisplayed] = useState(false);

  return (
    <View>
      <View style={styles.emailInputWrapper}>
        <TextInput
          style={styles.emailInput}
          placeholder={placeholder}
          keyboardType="default"
          secureTextEntry={!passwdIsDisplayed}
          value={value}
          onChangeText={onChangeText}
        />
        <View style={styles.emailIcon}>{icon()}</View>
      </View>
      <View
        style={{
          position: "absolute",
          top: 25,
          right: 12,
          backgroundColor: "transparent"
        }}
      >
        {passwdIsDisplayed ? (
          <EyeVisibleIcon
            onPress={() => setPasswdIsDisplayed(!passwdIsDisplayed)}
          />
        ) : (
          <EyeInvisibleIcon
            onPress={() => setPasswdIsDisplayed(!passwdIsDisplayed)}
          />
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  emailInputWrapper: {
    position: "relative"
  },
  emailInput: {
    borderColor: "#d1d3d9",
    fontFamily: "SatoshiRegular",
    borderWidth: 1,
    marginVertical: 24,
    paddingVertical: 15,
    paddingLeft: 55,
    paddingRight: 40,
    borderRadius: 30,
    fontSize: 17
  },
  emailIcon: { position: "absolute", top: 43, left: 22 }
});

export default PasswordInputWithIcon;
