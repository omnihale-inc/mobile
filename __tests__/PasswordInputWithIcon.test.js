import renderer from "react-test-renderer";

import PasswordInputWithIcon from "../components/PasswordInputWithIcon";

it("render password input securetextEntry as false", () => {
  const FakeComponent = jest.fn();
  const tree = renderer
    .create(
      <PasswordInputWithIcon
        icon={() => <FakeComponent />}
        onChangeText={jest.fn()}
        value="fake value"
        secureText={false}
        placeholder="fake value"
        eyeInvisibleOnPress={jest.fn()}
        eyeVisibleOnPress={jest.fn()}
        passwdIsDisplayed
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("render password input securetextEntry as true", () => {
  const FakeComponent = jest.fn();
  const tree = renderer
    .create(
      <PasswordInputWithIcon
        icon={() => <FakeComponent />}
        onChangeText={jest.fn()}
        value="fake value"
        secureText={false}
        placeholder="fake value"
        eyeInvisibleOnPress={jest.fn()}
        eyeVisibleOnPress={jest.fn()}
        passwdIsDisplayed={false}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
