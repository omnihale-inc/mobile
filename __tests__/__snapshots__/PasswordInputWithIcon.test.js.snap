// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render password input with icon 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "position": "relative",
        },
      ]
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[MockFunction]}
      placeholder="fake value"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#d1d3d9",
          "borderRadius": 30,
          "borderWidth": 1,
          "fontFamily": "SatoshiRegular",
          "fontSize": 17,
          "marginVertical": 24,
          "paddingLeft": 55,
          "paddingRight": 40,
          "paddingVertical": 15,
        }
      }
      value="fake value"
    />
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "left": 22,
            "position": "absolute",
            "top": 43,
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "position": "absolute",
          "right": 12,
          "top": 25,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <ViewManagerAdapter_ExpoImage
        contentFit="cover"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={60}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={[]}
        source={[]}
        style={
          {
            "height": 60,
            "width": 60,
          }
        }
        transition={null}
        width={60}
      />
    </View>
  </View>
</View>
`;
